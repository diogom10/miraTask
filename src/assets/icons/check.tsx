import * as React from 'react';
import Svg, {Path, SvgProps} from 'react-native-svg';

interface ICheckIconProps extends SvgProps {}

export const CheckIcon: React.FC<ICheckIconProps> = ({...props}) => (
  <Svg viewBox="0 0 512 512" {...props}>
    <Path d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm-38 312.38-80.6-89.57 23.79-21.41 56 62.22L350 153.46 374.54 174z" />
  </Svg>
);
