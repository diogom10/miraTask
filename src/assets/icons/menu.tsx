import * as React from 'react';
import Svg, {Circle, SvgProps} from 'react-native-svg';

interface IMenuIconProps extends SvgProps {}

export const MenuIcon: React.FC<IMenuIconProps> = ({...props}) => (
  <Svg viewBox="0 0 512 512" {...props}>
    <Circle cx={256} cy={256} r={48} />
    <Circle cx={256} cy={416} r={48} />
    <Circle cx={256} cy={96} r={48} />
  </Svg>
);
