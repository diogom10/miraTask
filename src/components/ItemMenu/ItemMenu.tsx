import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';

export interface ItemMenuProps {
  icon: React.ReactElement;
  title: string;
  action?: () => void;
  testID?: string;
}

export const ItemMenu = ({
  icon,
  title,
  action = () => null,
  testID = 'ItemMenu',
}: ItemMenuProps) => {
  return (
    <TouchableOpacity
      testID={`${testID}_btn_action`}
      style={styles.container}
      onPress={action}>
      {icon}
      <Text testID={`${testID}_title`} style={styles.text}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    marginHorizontal: 20,
    flexDirection: 'row',
    alignItems: 'center',
    minHeight: 20,
    marginBottom: 20,
    paddingBottom: 5,
    borderBottomWidth: 1,
    borderBottomColor: '#bdbfd3',
  },
  text: {
    fontSize: 15,
    fontWeight: '400',
    color: '#6b7093',
    marginLeft: 20,
  },
});
