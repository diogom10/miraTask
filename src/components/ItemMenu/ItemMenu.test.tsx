import React from 'react';
import {create, act} from 'react-test-renderer';
import {ItemMenu, ItemMenuProps} from './ItemMenu';
import {CheckIcon} from '../../assets/icons/check';

const props: ItemMenuProps = {
  title: 'Mark All tasks as Completed',
  icon: (
    <CheckIcon
      testID={`ItemMenu_icon`}
      width={20}
      height={20}
      fill={'#4270d4'}
    />
  ),
  testID: 'ItemMenu',
};

const component = create(<ItemMenu {...props} />);
const el_title = `${props.testID}_title`;
const el_icon = `${props.testID}_icon`;
const el_btn_action = `${props.testID}_btn_action`;

describe('ItemMenu Component', () => {
  test(`should render btn #${el_btn_action}`, async () => {
    const button = component.root.findByProps({testID: el_btn_action}).props;
    act(() => button.onPress());
  });
  test(`should render icon #${el_icon}`, async () => {
    const label = component.root.findByProps({testID: el_icon}).props;
    expect(label).toBeTruthy();
  });
  test(`should render label #${el_title}`, async () => {
    const label = component.root.findByProps({testID: el_title}).props;
    expect(label).toBeTruthy();
  });

  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
