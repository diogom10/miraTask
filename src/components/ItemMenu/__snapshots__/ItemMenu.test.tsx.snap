// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemMenu Component snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "borderBottomColor": "#bdbfd3",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "marginBottom": 20,
      "marginHorizontal": 20,
      "minHeight": 20,
      "opacity": 1,
      "paddingBottom": 5,
    }
  }
  testID="ItemMenu_btn_action"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="20"
    bbWidth="20"
    fill="#4270d4"
    focusable={false}
    height={20}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    testID="ItemMenu_icon"
    vbHeight={512}
    vbWidth={512}
    width={20}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4282544340,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M256 48C141.31 48 48 141.31 48 256s93.31 208 208 208 208-93.31 208-208S370.69 48 256 48zm-38 312.38-80.6-89.57 23.79-21.41 56 62.22L350 153.46 374.54 174z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      {
        "color": "#6b7093",
        "fontSize": 15,
        "fontWeight": "400",
        "marginLeft": 20,
      }
    }
    testID="ItemMenu_title"
  >
    Mark All tasks as Completed
  </Text>
</View>
`;
